### PR TITLE
[Test Modernization] Switch to using mountWithApp on ColorPicker tests.

### PR DESCRIPTION
**Pull request by lucabezerra**
### WHY are these changes introduced?

Modernizes tests to use the new modern framework (`{mountWithApp}` from `test-utilities`).

### WHAT is this pull request doing?

Updates tests using `{mountWithAppProvider}` from `test-utilities/legacy` to use `{mountWithApp}` from `test-utilities`

### CODE REVIEW

File: src/components/ColorPicker/tests/ColorPicker-ssr.test.tsx [hunk 1]
```diff
@@ -19,11 +18,15 @@ jest.mock('../../../utilities/target', () => ({
 
 describe('<ColorPicker /> Server-side only', () => {
   it('does not attach the touchmove handler to the window', () => {
-    const colorPicker = mountWithAppProvider(
+    const colorPicker = mountWithApp(
       <ColorPicker color={red} onChange={noop} />,
     );
 
-    colorPicker.find(Slidable).first().simulate('mousedown');
+    colorPicker!.find(Slidable)!.find('div')!.trigger('onMouseDown', {
```
Comment (BPScott) [inline src/components/ColorPicker/tests/ColorPicker-ssr.test.tsx:25]: colorPicker shall always be defined and will never be null so you don't need the first  `!` to manually assert that.

```suggestion
    colorPicker.find(Slidable)!.find('div')!.trigger('onMouseDown', {
```

File: src/components/ColorPicker/tests/ColorPicker.test.tsx [hunk 2]
```diff
@@ -13,26 +12,31 @@ const red = {
...
         const spy = jest.fn();
-        mountWithAppProvider(<ColorPicker color={red} onChange={spy} />)
-          .find(Slidable)
-          .at(SlidableType.BrightnessSaturation)
-          .simulate('mousedown');
+        const colorPicker = mountWithApp(
+          <ColorPicker color={red} onChange={spy} />,
+        );
+
+        colorPicker.findAll(Slidable)[SlidableType.Color].trigger('onChange', {
```
Comment (BPScott) [inline src/components/ColorPicker/tests/ColorPicker.test.tsx:29]: The enum feels really like a really weird use here. What do you think to removing it and using plain old numbers for indexes?

```suggestion
        colorPicker.findAll(Slidable)[0].trigger('onChange', {
```

File: src/components/ColorPicker/tests/ColorPicker.test.tsx [hunk 3]
```diff
@@ -91,46 +124,50 @@ describe('<ColorPicker />', () => {
...
 
-      colorPicker
-        .find(Slidable)
-        .at(SlidableType.BrightnessSaturation)
-        .simulate('mousedown');
-
-      const mouseMoveListener = colorPicker
-        .find(EventListener)
-        .filterWhere((listener) => listener.prop('event') === 'mousemove');
+      colorPicker!.find(Slidable)!.find('div')!.trigger('onMouseDown', {
```
Comment (BPScott) [inline src/components/ColorPicker/tests/ColorPicker.test.tsx:131]: No need for that first non-null assertion as colorPicker will never be null

```suggestion
      colorPicker.find(Slidable)!.find('div')!.trigger('onMouseDown', {
```

File: src/components/ColorPicker/tests/ColorPicker.test.tsx [hunk 4]
```diff
@@ -91,46 +124,50 @@ describe('<ColorPicker />', () => {
...
 
-      colorPicker
-        .find(Slidable)
-        .at(SlidableType.BrightnessSaturation)
-        .simulate('mousedown');
-
-      const touchMoveListener = colorPicker
-        .find(EventListener)
-        .filterWhere((listener) => listener.prop('event') === 'touchmove');
+      colorPicker!.find(Slidable)!.find('div')!.trigger('onMouseDown', {
```
Comment (BPScott) [inline src/components/ColorPicker/tests/ColorPicker.test.tsx:149]:
```suggestion
      colorPicker.find(Slidable)!.find('div')!.trigger('onMouseDown', {
```

File: src/components/ColorPicker/tests/ColorPicker.test.tsx [hunk 5]
```diff
@@ -91,46 +124,50 @@ describe('<ColorPicker />', () => {
...
     });
 
     it('prevents default for touchmove events if dragging the slider', () => {
-      const colorPicker = mountWithAppProvider(
+      const colorPicker = mountWithApp(
         <ColorPicker color={red} onChange={noop} />,
       );
 
-      colorPicker.find(Slidable).first().simulate('mousedown');
+      colorPicker!.find(Slidable)!.find('div')!.trigger('onMouseDown', {
```
Comment (BPScott) [inline src/components/ColorPicker/tests/ColorPicker.test.tsx:166]:
```suggestion
      colorPicker.find(Slidable)!.find('div')!.trigger('onMouseDown', {
```

**Review by BPScott — COMMENTED** [review]
Nice!

What do you think to removing that enum and using plain numbers for indexes?

There's a few places where you're using `!` to assert that colorPicker won't be null but typescript knows that already so you can remove it

**Comment by lucabezerra** [comment]
> Nice!
> 
> What do you think to removing that enum and using plain numbers for indexes?
> 
> There's a few places where you're using `!` to assert that colorPicker won't be null but typescript knows that already so you can remove it

@BPScott Thanks for the comments! I didn't like the enum either, but at the same time I thought it had some value in terms of giving names to indexes to make it clearer what they represented. I've removed it anyways :)

**Review by BPScott — APPROVED** [review]
:shipit:

**Pull request merged**